### PR TITLE
hotfix(medusa): Deleted region are missing on order retrieval

### PR DESCRIPTION
**Pull request by adrien2p**
WIP investigation

### CODE REVIEW

File: packages/medusa/src/repositories/order.ts [hunk 1]
```diff
@@ -25,7 +25,8 @@ export class OrderRepository extends Repository<Order> {
       Object.entries(groupedRelations).map(async ([_, rels]) => {
         return this.findByIds(entitiesIds, {
           select: ["id"],
-          relations: rels as string[],
+          relations: rels,
+          withDeleted: rels.includes("region"),
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/order.ts:29]: Wouldn't it be:
```
 Object.entries(groupedRelations).map(async ([rel, rels]) => {
        return this.findByIds(entitiesIds, {
          select: ["id"],
          relations: rels,
          withDeleted: rel === "region"
```

I believe `rels` will be all the additional relations added on `region` no?

Comment (olivermrbl) [reply packages/medusa/src/repositories/order.ts:29]: The passed relations: `["region", "region.[something]", "region.[something_else]"]`

Comment (adrien2p) [reply packages/medusa/src/repositories/order.ts:29]: I did the changes using topLevel but it is the same

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**